pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -0,0 +1,323 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import { superuser } from 'superuser';
+import { InstallProgressCB, MissingPackages, PackageManager, ProgressCB, ResolveError, InstallProgressType } from './packagemanager-abstract';
+
+let _dbus_client: cockpit.DBusClient | null = null;
+
+/**
+ * Get dnf5daemon D-Bus client
+ *
+ * This will get lazily initialized and re-initialized after dnf5daemon
+ * disconnects (due to a crash or idle timeout).
+ */
+function dbus_client() {
+    if (_dbus_client === null) {
+        _dbus_client = cockpit.dbus("org.rpm.dnf.v0", { superuser: "try", track: true });
+        _dbus_client.addEventListener("close", () => {
+            console.warn("dnf5daemon went away from D-Bus");
+            _dbus_client = null;
+        });
+    }
+
+    return _dbus_client;
+}
+
+// Reconnect when privileges change
+superuser.addEventListener("changed", () => {
+    if (_dbus_client)
+        _dbus_client.close();
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22397-224866fb-20250919-095321-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)